src/odemis/gui/cont/correlation.py

```diff
@@ -191,6 +217,12 @@ def add_streams(self, streams: list) -> None:
             if s in self._tab_data_model.streams.value or not isinstance(s, StaticStream):
                 continue
 
+            # if the user has loaded a rgb stream, assume it is meant to be a SEM stream
+            # (convert to 2D SEM stream, fix metadata, etc.)
+            if isinstance(s, RGBStream):
+                logging.debug(f"Converting RGB stream to SEM: {s.name.value}")
+                s = convert_rgb_to_sem(s)
+
```

Comment:
I agree this makes sense to only place it in the correlation tab, as it's a special case of what the user expectation, which is only valid for the correlation tab.
However, this is now called for every stream added (even from the localization tab). Also, it's a little inefficient, as it first creates a RGB stream, and then discards it for creating a new stream. Please move the whole code to `correlation_tab.load_stream()`. *Before* the call to `data_to_static_streams()`, you can check for each data, whether it is RGB, and if so, drop the the extra channels.

Reply:
I disagree, I think since it is essentially error handling it should happen as late as possible. The streams from the localization tab will never be rgb (unless something has gone very wrong) so there isn't really any extra work being done there. 
In terms of efficiency, moving it before data_to_static_streams would mean going through each data another time, and performing a similar check that happens in data_to_static_streams, rather than a simple comparison after its loaded which seems less efficient, even if every once in a while the stream needs to be re-created.  I also think there is a higher chance of making a bad assumption if we modify the data before data_to_static_streams.

Reply:
OK.